src/Collection/Collection.service.ts

```diff
@@ -446,13 +450,23 @@ export class CollectionService {
       return acc
     }, {} as Record<string, string>)
 
+    const slotUsageCheckHash = await getChequeMessageHash(
+      slotUsageCheque,
+      slotUsageCheque.third_party_id
+    )
+
+    const remoteCheque = await thirdPartyAPI.fetchReceiptById(
+      slotUsageCheckHash
+    )
```

Comment:
The remote cheque not existing is something that could happen? or should we throw here?

Reply:
It could happen if the curator approved the set of items and refreshed immediately, but the contract will fail as it will try to re-use the same cheque. It should be quite uncommon.

Reply:
oh ok, understood!